two-factor-authentication/verify-webhook/verify-webhook.5.x.rb

```diff
@@ -8,16 +8,9 @@ class CallbackVerifier
   def verify_callback(request, api_key)
     url = url_for(:only_path => false, :overwrite_params=>nil)
 
-    # Sort and join the parameters
-    parameter_string = request
-      .query_parameters
-      .to_a
-      .concat(request.request_parameters.to_a)
-      .sort
-      .map{ |p| p.join('=')}
-      .join('&')
-
-    parameter_string = URI.encode(parameter_string)
+    # Sort and join the parameters on Rails
+    json_params = JSON.parse(request.body.read)
+    parameter_string = json_params.to_query
```

Comment:
Are the parameters in the string sorted in alphabetical order?

Reply:
Good catch. I don't see the sort anywhere but it "magically" worked when I tested it with the Authy JSON body.
I will review it more in detail.

Reply:
It looks like on Rails `Hash#to_query` extension [sorts the query parameters](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/to_query.rb#L82) before joining them.

Reply:
Nice one! Gotta love magic :slightly_smiling_face: